### PR TITLE
 Use some sort of multiple select component for Tag input

### PR DESCRIPTION
**Pull request by lotrien**
To improve UX and let user intuitively add multiple tags, this commit
contains functionality that allows to add multiple tags by simply
clicking Enter button
Closes: #46

### CODE REVIEW

File: src/components/NewSnippet.jsx [hunk 1]
```diff
@@ -20,28 +20,35 @@ class NewSnippet extends React.Component {
       tags: [],
       syntax: '', // eslint-disable-line react/no-unused-state
     };
+    this.onKeyPress = (e) => {
+      if (e.which === 13) { e.preventDefault(); }
```
Comment (malor) [inline src/components/NewSnippet.jsx:24]: Is there a constant for that? If not let's at least document, that `13` is a keycode of `Enter`

File: src/components/NewSnippet.jsx [hunk 2]
```diff
@@ -20,28 +20,35 @@ class NewSnippet extends React.Component {
...
   }
 
   componentDidMount() {
     const { dispatch } = this.props;
-
     dispatch(actions.fetchSyntaxes);
   }
 
+  onTagAdded(tag) {
+    this.setState({ tags: [...this.state.tags, tag] });
```
Comment (malor) [inline src/components/NewSnippet.jsx:39]: nit: I think we currently do not enforce it at the API level, but maybe we should prevent duplicates of tags on the same snippet? @ikalnytskyi what do you think?

**Review by malor — APPROVED** [review]
I like the UX! And you can use not only Enter, but also Space or Tab to separate tags.

Overall looks good to me 👍  Small nits inline.

File: src/components/NewSnippet.jsx (hunk 2 above)
Comment (ikalnytskyi) [reply src/components/NewSnippet.jsx:39]: I think it makes sense. Will you file an issue for API to enforce that?

Comment (malor) [reply src/components/NewSnippet.jsx:39]: @ikalnytskyi do you want to reject requests with duplicates with `400 Bad Request`? Or just remove the duplicates before actually saving the snippet to the db?

Comment (ikalnytskyi) [reply src/components/NewSnippet.jsx:39]: @malor Honestly, I don't have strong opinion here. Both approaches have their own pros and cons. What do you think?

**Pull request merged**